sklearn/model_selection/_validation.py

```diff
@@ -779,6 +795,14 @@ def learning_curve(estimator, X, y, groups=None,
     return train_sizes_abs, out[0], out[1]
 
 
+def _shuffle_train_indices(cv_iter, shuffle, random_state):
```

Comment:
I think we had a little misunderstanding in the creation of this function. Can you please put it back inline? Thanks.